src/components/30-organisms/footer/index.js

```diff
@@ -0,0 +1,279 @@
+import styles from './index.scss';
+
+class AXAFooter extends LitElement {
+  static get tagName() {
+    return 'axa-footer';
+  }
+
+  static get styles() {
+    return css`
+      ${unsafeCSS(styles)}
+    `;
+  }
+
+  static get properties() {
+    return {
+      dynamic: { type: Boolean },
+    };
+  }
+
+  constructor() {
+    super();
+    this.dynamic = false;
+    this.onItemClick = () => {}; // Only for react
+    this._accordionActiveIndex = -1;
+  }
+
+  updated() {
+    this._removeEmptyListElements();
+  }
+
+  _removeEmptyListElements() {
+    this.shadowRoot
+      .querySelectorAll('.o-footer__main-content-panel-list-item')
+      .forEach(el => {
+        const label = el.querySelector('slot').assignedNodes()[0];
+        // Second part of IF-statement is an IE11 workaround
+        if (this._listElementHasNoContent(label)) {
+          el.style.display = 'none';
+        }
+      });
+  }
+
+  _listElementHasNoContent(label) {
+    return !label || label.toString() === '[object Text]';
+  }
+
+  _handleLinkClick = ev => {
+    if (this.dynamic) {
+      ev.preventDefault();
+
+      const href = this._extractNestedHref(ev);
+
+      if (href) {
+        this.onItemClick(href);
+        this.dispatchEvent(
+          new CustomEvent('axa-footer-click', {
+            detail: href,
+            bubbles: true,
+            cancelable: true,
+          })
+        );
+      }
+    }
+  };
+
+  _extractNestedHref(ev) {
+    let element;
+    if (!ev.target || !ev.target.href) {
+      element = ev;
+      while (!element.href) {
+        if (element.target && element.target.parentNode)
+          element = element.target.parentNode;
+        else if (element.parentNode) element = element.parentNode;
+        else return undefined;
+      }
+    }
+    return element === undefined ? ev.target.href : element.href;
+  }
+
+  _handleAccordionClick = index => {
+    this._accordionActiveIndex =
+      index === this._accordionActiveIndex ? -1 : index;
+    this.requestUpdate();
+  };
+
+  render() {
+    return html`
+      <footer class="o-footer">
+        <axa-container>
+          <div class="o-footer__content">
+            <div class="o-footer__collection">
+              <div class="o-footer__main">
+                <slot
+                  name="column-0-title-desktop"
+                  class="o-footer__title-desktop"
+                ></slot>
+                <button
+                  class="o-footer__accordion-button"
+                  @click="${() => this._handleAccordionClick(0)}"
+                >
+                  <slot name="column-0-title" class="o-footer__title"></slot>
+                  <span
+                    class="o-footer__accordion-button-caret o-footer__accordion-button-caret${this
+                      ._accordionActiveIndex === 0
+                      ? '--open'
+                      : ''}"
+                  >
+                    ${svg([CaretSvg || ''])}
```

Comment:
This expression should be lifted to the beginning of render() and given a proper name

Reply:
yes!

Reply:
I can do that, but I don't see a reason. Isn't that readable?

Reply:
done